presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -114,6 +114,13 @@ public static Builder builder()
         return new Builder();
     }
 
+    /**
+     * If one of the estimates below is unspecified, the default "unknown" estimate value
+     * (represented by floating point NaN value) may cause the resulting symbol statistics
+     * to be "unknown" as well.
+     * Make sure to specify all the estimates below to allow cost-based query optimization.
+     * @see SymbolStatsEstimate
+     */
```

Comment:
I am sorry, because I am a bit biased as I know this code a bit, but to me this comment looks obvious. 

 - Any operation on an unknown value leads to an unknown result, we could say unknown propagates. It is like `null` in SQL.

 - > If one of the estimates below is unspecified, the default "unknown" estimate value

This is already written in Java two lines below.

I understand you had hard issue to debug, but reaching here to read this comment is the same as reaching here to read the code below. What do you think?

Reply:
Javadoc is public documentation and is shown by IDEs and in the generated Javadoc (you don't have to look at the source code to find it). Users of a public API shouldn't have to read the internal implementation to understand it.